xarray/core/groupby.py

```diff
@@ -771,10 +771,7 @@ def _flox_reduce(
             # the bin edge labels have a default precision of 3
             # reassign to fix that.
             assert self._full_index is not None
-            new_coord = [
-                pd.Interval(inter.left, inter.right) for inter in self._full_index
-            ]
-            result[self._group.name] = new_coord
+            result[self._group.name] = self._full_index
```

Comment:
@benbovy I am overwriting an existing index. What's the best way to do this?

Reply:
If `self._group.name` is always a dimension name and `self._full_index` is always a single pandas index, then  `result[self._group.name] = self._full_index` should work fine I think (it will probably make a copy of the index).

Otherwise it could be more complicated.

In general one would first wrap the pandas index with `PandasIndex` (or `PandasMultiIndex`), then get the corresponding coordinate variables(s) with `PandasIndex.create_variables()`, and then assign both the index and the variables to the Dataset / DataArray, either directly via `._indexes`, `._variables` or `._coords`, or via `._overwrite_indexes()` for the more advanced cases.

Reply:
I can use the current version here.

> Otherwise it could be more complicated.

Shall we create a simple `_assign_single_pandas_index` helper method to avoid the copy?

Reply:
If keeping the same index object in the result is important, yes, otherwise a copy is fine as I think it has almost no impact on performance.